### ISSUE TITLE
[BUG] Dynamic routes have higher priority over static routes

### ISSUE THREAD

**Issue by Joshua-Tu** [oas-tools/oas-tools#355]
## **Describe the bug**
I have two API endpoint definition (**/samplePath/_health** and **/samplePath/{id}**)
```
{
    "openapi": "3.0.0",
    "info": {
        "title": "Sample Service API",
        "description": "All about Sample Service",
        "version": "1.0"
    },
    "paths": {
        "/samplePath/_health": {
            "get": {
                "x-router-controller": "sampleController",
                "x-router-handle-subpaths": false,
                "operationId": "getSampleHealth",
                "tags": ["Api Health"],
                "description": "API to check health of the service",
                "responses": {
                    "200": {
                        "description": "Successful request.",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/successResponse"
                                }
                            }
                        }
                    },
                    "default": {
                        "description": "Invalid request.",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Error"
                                }
                            }
                        }
                    }
                }
            }
        },
        "/samplePath/{id}": {
            "get": {
                "x-router-controller": "sampleController",
                "x-router-handle-subpaths": false,
                "operationId": "getSample",
                "description": "get sample by ID",
                "tags": ["Template"],
                "parameters": [
                    {
                        "name": "id",
                        "in": "path",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful request.",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/successResponse"
                                }
                            }
                        }
                    },
                    "default": {
                        "description": "Invalid request.",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Error"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    ...
}
```
But when I call `/samplePath/_health`. I got this error: `{"error":"TypeError: Cannot read property 'forEach' of undefined"}`
Which is thrown from here: oas-validator.js
## **Screenshots**
<img width="884" alt="image" src="https://user-images.githubusercontent.com/47619573/207327421-0ebb169e-93bf-46e1-9116-fe63cdeceb14.png">

I added some console.logs in the oas-validator.js and oas-params.js
<img width="783" alt="image" src="https://user-images.githubusercontent.com/47619573/207328376-02fcd43d-658f-47d4-80cb-4e3080a54a0e.png">

Then I got the following info.
Seems like oas tools treats the /samplePath/_health endpoint's _health as a path param.

```
🚀 ~ file: oas-params.js:13 ~ OASParams ~ methodParams undefined
🚀 ~ file: oas-params.js:14 ~ OASParams ~ pathParams undefined
🚀 ~ file: oas-params.js:13 ~ OASParams ~ methodParams 
[
    {
         name: 'id',
         in: 'path',
         required: true,
         schema: { type: 'string' }
    }
]
🚀 ~ file: oas-params.js:14 ~ OASParams ~ pathParams undefined
[12/12/2022 23:17:40.874 +00:00] [info] : Requested GET /templates/_health
🚀 ~ file: oas-validator.js:52 ~ OASRequestValidator ~ res.locals.oas.params { id: '_health' }
[12/12/2022 23:17:40.875 +00:00] [error] : TypeError: Cannot read property 'forEach' of undefined
```

## **Desktop (please complete the following information):**
 - OS: Ventura
 - Version: 13.0.1

## **Additional context**
@alesancor1 mentioned 'This is a problem with the router, static routes should take preference over dynamic routes.'

**Comment by Joshua-Tu** [comment]
Hi, @alesancor1 noticed that there are still no replies on this one, is it going to be fixed in the next release?

**Comment by Joshua-Tu** [comment]
@alesancor1 Hi, could I know when this bug will be fixed?

**Comment by alesancor1** [comment]
This issue has been fixed at eaf73a03f5129ee3a818418f1b2bef5872e2458b. The problem was caused by the way middlewares were being registered, since each middleware was being registered for each path. I modified the code so every middleware is registered for a single path before moving to the next path for registration.

This way, the middleware chain now follows express logic. Static routes should be placed before dynamic routes in OAS Document in order for them to take precedence. The order they are registered in express is the same you write in the OAS Document.
